src/app/modules/data-selection/components/patient-selection/patient-selection.component.ts

```diff
@@ -1,25 +1,33 @@
-import { Component, OnInit } from '@angular/core';
+import {Component, OnInit} from '@angular/core';
 import {ResourceService} from "../../../shared/services/resource.service";
+import {WorkflowService} from "../../../shared/services/workflow.service";
 
 @Component({
   selector: 'patient-selection',
   templateUrl: './patient-selection.component.html',
   styleUrls: ['./patient-selection.component.css'],
-  providers: [ResourceService]
+  providers: [ResourceService, WorkflowService]
 })
 export class PatientSelectionComponent implements OnInit {
+  patientCount: number;
 
-  constructor(private resourceService: ResourceService) { }
+  constructor(private resourceService: ResourceService, private workflowService: WorkflowService) {
+  }
 
   ngOnInit() {
+    this.patientCount = 0;
   }
 
   runPatientQuery() {
-    // let patients: Patient[];
+
+    let currentWorkflow = this.workflowService.getCurrentWorkflow();
+
     console.log('run patient query');
     this.resourceService.getPatients().subscribe(
-      patients => {
-        console.log('patients: ', patients);
+      patientsObj => {
+        console.log('patients: ', patientsObj);
+        currentWorkflow.setPatients(patientsObj['patients']);
+        this.patientCount = currentWorkflow.getPatients().length;
       },
       err => {
         console.log(err);
```

Comment:
Maybe console.warn or console.error is preferred when logging an error instead of console.log